### DISCUSSION TITLE
Random symbolic simulation trick ?

### DISCUSSION THREAD

**Discussion by MathieuMarchand** [UPPAALModelChecker/UPPAAL-Meta#229] [Q&A] [answered]
I would like to run the symbolic simulator to see how it evolves, but since the random simulation takes one of the available transition with equal probability, the simulation doesn't do anything useful.

Is there a way to change these probabilities without having to duplicate edges, to have for example an edge -->goNext with p=0.99 and an edge -->abort with p=0.01?

**Comment by thorulf4** [comment, marked as answer]
There is no nice way of doing this for symbolic simulation, you could hack together some kind of probability by controlling the ratio of edges, but I would would discourage this approach.

If possible I would recommend switching to the concrete simulator, as it provides a way to give probabilistic weights to branching edges. 
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/10059450/e7970117-933e-446b-94c2-b73a361568c4)
Do note that probabilities can only be set on the outgoing side of a branching edge

For this to to work you would have to switch from random simulation to stochastic by right clicking the random button in the concrete simulator.
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/10059450/94277e0d-b509-48a4-b168-f413c780759d)

I hope this helps, if any questions feel free to ask.

**Reply by MathieuMarchand** [reply]
I see, thank you for the fast response ^^

**Comment by MathieuMarchand** [comment]
![image](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/44498497/20e3a2ce-a532-4e21-a42d-c4944b569c01)
Do you have to do something special to activate the stochastic option ?

**Reply by thorulf4** [reply]
I can't think of any reasons why Stochastic wouldn't be available here. What version of Uppaal are you using?

**Reply by MathieuMarchand** [reply]
UPPAAL-5.0.0-WIN64

**Reply by MathieuMarchand** [reply]
It seems something in my system disables the stochastic option. I tried on a new project and it is enabled. I'll investigate a bit

**Reply by MathieuMarchand** [reply]
I found the issue, you cannot have channels and execute stochastic simulation

**Reply by thorulf4** [reply]
This happens because SMC only supports broadcast channels. However beware that broadcast channels have slightly different semantics.
You can read about broadcast channels here:
https://docs.uppaal.org/language-reference/system-description/templates/edges/#synchronisations
